### PR TITLE
#MySQL_Tasks. dananaprey

### PR DESCRIPTION
**Pull request by dananaprey**
https://travis-ci.org/github/dananaprey/db-assignments/builds/710355907

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -281,7 +466,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
```
Comment (AlexMokin) [inline task/sql-tasks.js:467]: try to update this sql to run faster than 110ms

File: task/sql-tasks.js [hunk 2]
```diff
@@ -281,7 +466,26 @@ async function task_1_21(db) {
...
+            INNER JOIN Orders ON Orders.CustomerID = ac.CustomerID
+            INNER JOIN OrderDetails ON OrderDetails.OrderID = Orders.OrderID
+            INNER JOIN Products ON Products.ProductID = OrderDetails.ProductID
+        WHERE OrderDetails.UnitPrice = (
+            SELECT 
+                MAX(OrderDetails.UnitPrice) 
+            FROM Customers bc
+                INNER JOIN Orders ON Orders.CustomerID = bc.CustomerID
+                INNER JOIN OrderDetails ON OrderDetails.OrderID = Orders.OrderID
+            WHERE bc.CompanyName = ac.CompanyName
```
Comment (AlexMokin) [inline task/sql-tasks.js]: you're lookin for max price of the customer, company name doesn't uniquely identify customer

File: task/sql-tasks.js [hunk 3]
```diff
@@ -281,7 +466,26 @@ async function task_1_21(db) {
...
+            OrderDetails.UnitPrice AS 'PricePerItem'
+        FROM Customers ac
+            INNER JOIN Orders ON Orders.CustomerID = ac.CustomerID
+            INNER JOIN OrderDetails ON OrderDetails.OrderID = Orders.OrderID
+            INNER JOIN Products ON Products.ProductID = OrderDetails.ProductID
+        WHERE OrderDetails.UnitPrice = (
+            SELECT 
+                MAX(OrderDetails.UnitPrice) 
+            FROM Customers bc
+                INNER JOIN Orders ON Orders.CustomerID = bc.CustomerID
```
Comment (AlexMokin) [inline task/sql-tasks.js]: add aliases
image

File: task/sql-tasks.js [hunk 4]
```diff
@@ -106,7 +150,16 @@ async function task_1_6(db) {
  *
  */
 async function task_1_7(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT
+            DISTINCT a.EmployeeID AS 'EmployeeId',
+            CONCAT(a.FirstName, ' ', a.LastName) AS 'FullName',
+            IF(a.reportsTo, CONCAT(b.FirstName, ' ', b.LastName), '-') AS 'ReportsTo'
+        FROM Employees a, Employees b
```
Comment (AlexMokin) [inline task/sql-tasks.js]: try to implement this with left join

**Comment by dananaprey** [comment]
https://travis-ci.org/github/dananaprey/db-assignments/builds/717203746

**Comment by IlyaMokin** [comment]
Looking good, thanks! The PR is accepted

**Pull request closed without merging**